### PR TITLE
Bugfix/a bunch

### LINKED ISSUE #175: Clear question options when closing the creator modal
Right now it shallow copies the blank question when it should deep copy to include the question options

### LINKED ISSUE #177: Question preview doesn’t show the folder name
Next to the folder icon

### LINKED ISSUE #178: Question type not displayed in the content tree

### LINKED ISSUE #179: Content tree “Title” and “Question Type” should be stickied to the top

### LINKED ISSUE #180: Create Modal form requires tab indexing

### LINKED ISSUE #181: selecting an empty folder in present modal causes error smh my head
**Describe the bug**
selecting an empty folder in present modal causes error smh my head

**To Reproduce**
1. selecting an empty folder in present modal
2. smh
3. my head

**Expected behavior**
selecting an empty folder in present modal does not cause error

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Linux Mint 19.3 Cinnamon 64-bit
 - Browser: Chrome
 - Version: 83.0.4103.61 (Official Build) (64-bit)

**Smartphone (please complete the following information):**
 - Lmao imagine using smartphone xdddd
 
**Additional context**
Hurry up and fix it I already promised my son this would be working, he also has cancer

### PR DESCRIPTION
**Pull request by weining-li**
Resolves #181 
Resolves #180 
Resolves #179 
Resolves #178 
Resolves #177 
Resolves #175

### CODE REVIEW

**Review by kennygperez — APPROVED** [review]
LGTM

File: backend/src/@types/models/Question.d.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ interface QuestionAttributes {
 	question: string;
 	participationPoints: number;
 	correctnessPoints: number;
+	type: string;
```
Comment (kennygperez) [inline backend/src/@types/models/Question.d.ts:11]: This should be an enum

so TS can minimize human error

File: backend/src/models/Question.ts [hunk 2]
```diff
@@ -37,4 +37,9 @@ export const Question: ModelDefined<
 		allowNull: false,
 		defaultValue: 0.5,
 	},
+	type: {
+		type: DataTypes.STRING,
+		allowNull: false,
+		defaultValue: 'Mult Choice',
```
Comment (kennygperez) [inline backend/src/models/Question.ts:43]: Could use enum value

**Review by Goff-Davis — CHANGES_REQUESTED** [review]
Since kenny requested changes I'm putting this here until I can review it as well

**Comment by kennygperez** [comment]
i didnt its just stuff i would have liked to see :<

**Comment by kennygperez** [comment]
@Goff-Davis ^

File: frontend/src/components/creator-modal/creator-edit/creator-edit.tsx [hunk 3]
```diff
@@ -162,7 +196,7 @@ const CreatorEdit = (): ReactElement => {
             <input
```
Comment (Goff-Davis) [inline frontend/src/components/creator-modal/creator-edit/creator-edit.tsx:196]: The add answer choice button still needs to be given a tab index

File: frontend/src/store.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ const init = {
   questionNumber: 0,
   closedQuestions: closedQuestions,
   openExportModal: false,
-  currentQuestionInfo: baseQuestionInfo,
+  currentQuestionInfo: JSON.parse(JSON.stringify(baseQuestionInfo)),
```
Comment (Goff-Davis) [inline frontend/src/store.js:33]: Doesn't doing `JSON.parse(JSON.stringify(stuff))` just give you what it was originally?

**Review by Goff-Davis — CHANGES_REQUESTED** [review]
One change and one question

File: frontend/src/store.js (hunk 4 above)
Comment (weining-li) [reply frontend/src/store.js:33]: yes. 

previously the `baseQuestionInfo` and `currentQuestionInfo` were both referencing the same object, which causes the changes done to the answer options array in `currentQuestionInfo` be reflected in `baseQuestionInfo`. 

this way it's creating a copy of the `baseQuestionInfo` object instead which stops them from being linked.

Comment (weining-li) [reply frontend/src/store.js:33]: actually now that you commented I realized I also need to do this in reset

File: frontend/src/components/creator-modal/creator-edit/creator-edit.tsx (hunk 3 above)
Comment (kennygperez) [reply frontend/src/components/creator-modal/creator-edit/creator-edit.tsx:196]: Why are you guys messing with the tab index? the default usually works very well

File: frontend/src/store.js (hunk 4 above)
Comment (kennygperez) [reply frontend/src/store.js:33]: Yeah idk whats doing on here

File: frontend/src/components/creator-modal/creator-edit/creator-edit.tsx (hunk 3 above)
Comment (weining-li) [reply frontend/src/components/creator-modal/creator-edit/creator-edit.tsx:196]: some of the elements are divs

File: frontend/src/store.js (hunk 4 above)
Comment (kennygperez) [reply frontend/src/store.js:33]: @weining-li then using `Object.assign({}, whatever)`

Comment (Goff-Davis) [reply frontend/src/store.js:33]: Ok cool

**Review by Goff-Davis — APPROVED** [review]

File: frontend/src/store.js (hunk 4 above)
Comment (Goff-Davis) [reply frontend/src/store.js:33]: > @weining-li then using `Object.assign({}, whatever)`

I personally would have used `Object.assign`. But both do the same thing so 🤷

Comment (weining-li) [reply frontend/src/store.js:33]: LJ said we need to deep copy but `Object.assign` is shallow copy.

Comment (Goff-Davis) [reply frontend/src/store.js:33]: U right. I thought `Object.assign` did deep copy but it doesn't

**Review by ljoks — CHANGES_REQUESTED** [review]
- don't tab thru the `Correct Answer` and `X` buttons of the question options, just go straight to the next text input
- fix folder name spacing in the content tree
- `.tab-buttons` height is messed up for me. even if it's just my instance for some reason, might as well put an explicit height on it

**Comment by Goff-Davis** [comment]
>     * `.tab-buttons` height is messed up for me. even if it's just my instance for some reason, might as well put an explicit height on it

It seems like flex-shrink isn't being applied consistently for some reason. A fix is to add the height of .tab-buttons to be min-content and to change the height of creator-header to be 5.6rem, instead of 6

**Comment by Goff-Davis** [comment]
Never mind, `9bb7093` works to fix it too

**Review by ljoks — APPROVED** [review]

**Pull request merged**